### DISCUSSION TITLE
Pessimistress (and visgl team) QA

### DISCUSSION THREAD

**Discussion by jessekrubin** [visgl/deck.gl#6775] [Q&A]
As far as I can tell @Pessimistress is the deck.gl/react-map-gl authority (correct me if I am wrong), so this is a discussion thread aimed at asking Pessimistress and the visgl team misc questions. 

### First Questions:

- What is the future of react-map-gl with respect to nebula-gl now that react-map-gl could support mapbox-gl-draw?
- What is the roadmap of deckgl + react-map-gl w/ respect to context-providing (which is no longer supported by react-map-gl 7)?
- Does deck.gl plan on moving the src to typescript? and if so when? It seems like a migration is slowly happening, so just checking in here...
- How can I create an svg-layer without doing `encodeUriComponent` on an svg string and putting the encoded image in an icon layer?
- Is there a way to why do my iconlayers occasionally turn black and opaque? This happens when Icons update rapidly for me.

**Comment by ibgreen** [comment]
> What is the future of react-map-gl with respect to nebula-gl now that react-map-gl could support mapbox-gl-draw?

nebula.gl is primarily a set of high-performance deck.gl "editable layer" modules, created independently of react-map-gl and to support editing of much larger datasets and with more features than map-gl-draw. Whether maintaining nebula.gl's react-map-gl adapter still makes sense may be a good question for the nebula.gl maintainers (though I have not heard this being discussed), but nebula.gl as a performance-focused, basemap-library-independent editing solution for deck.gl is not going away.

> Does deck.gl plan on moving the src to typescript? 

Yes

> and if so when? It seems like a migration is slowly happening, so just checking in here...

As soon as we can get it ready, which should be fairly soon. Note that a massive amount of work has been done on upgrading all the underlying frameworks to Typescript, so we are at the tail end of a much bigger effort than you may realize. This work is being done by maintainers on our free time, so needless to say, contributions are welcome :)

> How can I create an svg-layer without doing encodeUriComponent on an svg string and putting the encoded image in an icon layer?

What does this mean? You don't want to your SVGs to be converted bitmaps / icons? Are you looking for a WebGL rendererer for SVGs?

> Is there a way to why do my iconlayers occasionally turn black and opaque? This happens when Icons update rapidly for me.

Not really a discussion topic? Sounds more like a bug report where we typically ask users to provide a code pen to replicate the issue.

**Reply by Pessimistress** [reply]
Adding on to the TypeScript roadmap: we are currently targeting a "preview" release in v8.8, meaning all core/layers code will be fully typed, but the types will not be exposed by default. Users who are willing to try the types will be able to import them with a minimal effort. This is so that we can collect feedback and improve the types interface, especially for advanced use cases, before they are officially released.

**Comment by jessekrubin** [comment]
@Pessimistress @ibgreen I got some more questions for yall!

- What are the most common deck/react-map-gl anti patterns yall see/encounter?
- Why aren't the modules that have been converted to ts distributed with types? and/or at the very least why don't they (the ts modules) have a a `"types"` field in their `package.json`'s?

**Reply by Pessimistress** [reply]
Types are not exported because if we do, you don't have the option to opt out. A partially typed library will hard break your application if you are already using TypeScript.

**Reply by jessekrubin** [reply]
@Pessimistress how do I opt in if I wanted to give it a try? reference the raw src as types? FWIW: I would be down to give it a go and report back. I have been using the Dan Marshall typings for a bit now and they are mostly p good.

**Reply by Pessimistress** [reply]
Please follow https://github.com/visgl/deck.gl/issues/5589 for any updates.
